File: string/jewels-and-stones/jewels_and_stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewels_count = 0
        for s in stones:
            if s in jewels:
                jewels_count = jewels_count + 1
        return jewels_count

s = Solution()
print(s.numJewelsInStones("aA","aAAbbbb")) 

File: string/jewels-and-stones/jewels-and-stones.ts
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels: string, stones: string) {
  let jewelsCount = 0;
  for (let s of stones) {
    if (jewels.includes(s)) {
      jewelsCount++;
    }
  }
  return jewelsCount;
};

/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStonesReg = function (jewels: string, stones: string) {
  let stonesLength = stones.length;
  let jewelsReg = eval(`/${jewels.split('').join('|')}/g`);
  stones = stones.replace(jewelsReg, '');
  return stonesLength - stones.length;
};

